int16 posA
int16 posB
uint8 laser
